### PR TITLE
deal with old and new guides

### LINKED ISSUE #46: Package unable to load (due to the update of `ggplot2`?)
I'm using `ggplo2` 3.4 and when librarying `ggside`, it gives the following message:

```r
> library(ggside)
Error: package or namespace load failed for ‘ggside’:
 object ‘guide_gengrob’ is not exported by 'namespace:ggplot2'
```

I observed a similar issue [here](https://stackoverflow.com/questions/74394148/is-not-an-exported-object-from-namespaceggplot2) for another package, just fyi.

### PR DESCRIPTION
**Pull request by teunbrand**
TL;DR: This PR aims to fix #46.

Hi Justin,

We have been preparing a new release of ggplot2 and during a reverse dependency check, it became apparent that the prospective ggplot 3.5.0 would  break ggside.
Given that other packages also import ggside we felt it was prudent to help out to transition to the new version.
The culprit is that ggplot2 has changed the guide system, see https://github.com/tidyverse/ggplot2/pull/4879.
This PR handles the new guide infrastructure and should work with both the current and the development version of ggplot2.
To test the code changes with the release candidate, you can install it with the code below:

```r
remotes::install_github("tidyverse/ggplot2", ref = remotes::github_pull("5592"))
```

The release of ggplot2 3.5.0 is scheduled at the end of January / early Februari. The progress can be tracked in https://github.com/tidyverse/ggplot2/issues/5588.
We are hoping that this PR might help ggside be ready around that time to soften the impact on downstream packages.

### CODE REVIEW

File: R/sideCoord--utils.R [hunk 1]
```diff
@@ -23,9 +29,20 @@ guide_for_position <- function (guides, position)
...
-  guide_gengrob(guide, theme)
+  if (inherits(guides, "Guides")) {
+    pair <- guides$get_position(position)
+    # # To use new guide system activate the line below
+    # pair$guide$draw(theme, params = pair$params)
+    # # And deactivate the line below up until the `} else {` line
+    if (inherits(pair$params$angle, "waiver")) {
+      pair$params$angle <- NULL
+    }
+    guide_gengrob.ggside_axis(pair$params, theme)
```
Comment (teunbrand) [inline R/sideCoord--utils.R:40]: Using the new system would result in some visual changes related to the spacing around axes.
I didn't fully understand where this change would come from, so I retained the old drawing method.
If you want to switch to the new guide system, you can use `pair$guides$draw()` instead.

Comment (jtlandis) [reply R/sideCoord--utils.R:40]: I appreciate that you have taken the time to look into correcting these. I have been working on bringing `ggside` up to speed with `ggplot2 3.5.0` on the dev branch. I do not know which branch you were testing against, but my guess would be the `main` branch.

The `dev` branch should be compatible with `ggplot2`, but it isn't in a state where I'm ready to release. I've been putting a majority of my refactoring into `dev-refactor` which will eventually be merged to `dev` and then `main`.

I think in the future I will follow `ggplot2`'s coding strategy and leave `main` as the development branch that is out of sync with CRAN.

I have switched to the new guide system where relevant in `ggside` and just ran `ggside`'s tests against

```r
remotes::install_github("tidyverse/ggplot2", ref = remotes::github_pull("5592"))
```

Only about 12 tests out of 275 had failures in that there was more spacing given to the Y axis title when the axis was plotted on the right. I for one can live with these changes 👍 

Thanks again for your diligent work!

**Comment by teunbrand** [comment]
Apologies, I hadn't noticed the dev branches and all the work you had put in already. That is amazingly pro-active.
I was indeed testing against the main branch.

> Only about 12 tests out of 275 had failures in that there was more spacing given to the Y axis title when the axis was plotted on the right. 

This might be expected as the previous right y-axis title's `vjust` was doing the opposite as it should (see https://github.com/tidyverse/ggplot2/pull/5629), so I wouldn't see these as true failures.

If this PR is redundant, please feel free to close :)

**Comment by teunbrand** [comment]
> The dev branch should be compatible with ggplot2, but it isn't in a state where I'm ready to release.

We've settled on a release date for ggplot2 3.5.0 at February 12th. If it is unfeasible to get your big feature release out before then, might I gently and kindly put the notion out there that a small, targeted hotfix would be beneficial to minimise breakages? (I know this is ggplot2's fault for causing the breakages in the first place) If you'd like me to amend or retarget this PR to another branch, don't hesitate to ask :)

**Comment by jtlandis** [comment]
I have just submitted my package version, sorry for the radio silence, I wasn't sure if I could release a new version since my package depended on 3.5.0 to have CI tests pass as expected. 

I am going to close this PR. Thanks again for everything!
